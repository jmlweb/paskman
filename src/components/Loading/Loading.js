import React from 'react';
import LoadingWrapper from './LoadingWrapper/LoadingWrapper';
import LoadingMain from './LoadingMain/LoadingMain';
import LoadingGroups from './LoadingGroups/LoadingGroups';
import { LoadingLeftGroup, LoadingRightGroup } from './LoadingGroup/LoadingGroup';
import { LoadingPrimaryItem, LoadingSecondaryItem } from './LoadingItem/LoadingItem';
import LoadingSVG from './LoadingSVG/LoadingSVG';
import LoadingMsg from './LoadingMsg/LoadingMsg';

const Loading = ({ text }) => (
  <LoadingWrapper>
    <LoadingMain>
      <LoadingGroups>
        <LoadingLeftGroup>
          <LoadingPrimaryItem />
          <LoadingPrimaryItem />
          <LoadingPrimaryItem />
        </LoadingLeftGroup>
        <LoadingRightGroup>
          <LoadingSecondaryItem />
          <LoadingSecondaryItem />
          <LoadingSecondaryItem />
        </LoadingRightGroup>
      </LoadingGroups>
      <LoadingSVG />
    </LoadingMain>
    {text && <LoadingMsg>{text}</LoadingMsg>}
  </LoadingWrapper>
);

export default Loading;
