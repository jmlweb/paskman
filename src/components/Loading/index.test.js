import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Loading from './';

configure({ adapter: new Adapter() });

describe('Loading', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<Loading />);
  });
  it('Render the connected component', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });
  it('Matchs snapshot', () => {
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
