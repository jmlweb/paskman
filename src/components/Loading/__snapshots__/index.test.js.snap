// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Matchs snapshot 1`] = `
<div>
  Loading
</div>
`;
